### DISCUSSION TITLE
how to use python package ”impala“ to connect kyuubi ha server.

### DISCUSSION THREAD

**Discussion by ooeeerrtt** [apache/kyuubi#4434] [Q&A]
how to use python package ”impala“ to connect kyuubi ha server.

**Comment by pan3793** [comment]
"impala" or "impyla"?
Suppose you are talking about ["impyla"](https://github.com/cloudera/impyla), Kyuubi's THRIFT_BINARY protocol exposes the HiveServe2-compatible endpoint, so the question is does "impyla" support connect  to HiveServer2? If yes, it should work w/ Kyuubi, in same way.

**Reply by ooeeerrtt** [reply]
if i writer like:

```
from impala.dbapi import connect
conn = connect(host = 'kyuubi_host',
                 port = kyuubi_port,
                 auth_mechanism = 'PLAIN',
                 user = 'hadoop',
                 database = '**')
```
it's ok. but it's still use simple server

if i try ro connet ha server,i write like:
```
from impala.dbapi import connect
conn = connect(host = 'zookeeper_host',
                 port = 2181,
                 auth_mechanism = 'PLAIN',
                 user = 'hadoop',
                 database = '**')
```
it's  wrong.

thanks!

**Reply by pan3793** [reply]
I guess the impyla does not support HS2 HA protocol, I get nothing by searching on the impyla codebase/issue/PR using keywords "zookeeper", "ha".
